rust-src/concordium_base/src/web3id/did.rs

```diff
@@ -0,0 +1,480 @@
+            network: Network::Mainnet,
+            ty:      IdentifierType::ContractData {
+                address:    ContractAddress::new(index, subindex),
+                entrypoint: entrypoint.clone(),
+                parameter:  parameter.clone(),
+            },
+        };
+        assert_eq!(
+            format!("did:ccd:sci:{index}:{subindex}/{entrypoint}/{parameter}").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:mainnet:sci:{index}:{subindex}/{entrypoint}/{parameter}")
+                .parse::<Method>()?,
+            target
+        );
+        let s = target.to_string();
+        assert_eq!(s.parse::<Method>()?, target);
+        assert_eq!(
+            format!("did:ccd:testnet:sci:{index}:{subindex}/{entrypoint}/{parameter}")
+                .parse::<Method>()?,
+            Method {
+                network: Network::Testnet,
+                ..target
+            }
+        );
+        assert!(
+            format!("did:ccd:testnet:sci:{index}:{subindex}/{entrypoint}/{parameter}/ff")
+                .parse::<Method>()
+                .is_err()
+        );
+
+        // Section to test that omitting the subindex is OK when it is 0, and omitting
+        // the parameter is OK when it is empty.
+        let index = 37;
+        let subindex = 0;
+        let entrypoint = OwnedEntrypointName::new_unchecked("viewStuff".into());
+        let parameter = OwnedParameter::new_unchecked(vec![]);
+        let target = Method {
+            network: Network::Mainnet,
+            ty:      IdentifierType::ContractData {
+                address:    ContractAddress::new(index, subindex),
+                entrypoint: entrypoint.clone(),
+                parameter:  parameter.clone(),
+            },
+        };
+        assert_eq!(
+            format!("did:ccd:sci:{index}:{subindex}/{entrypoint}/{parameter}").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:sci:{index}/{entrypoint}/{parameter}").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:sci:{index}/{entrypoint}/").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:sci:{index}/{entrypoint}").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:mainnet:sci:{index}:{subindex}/{entrypoint}/{parameter}")
+                .parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:mainnet:sci:{index}/{entrypoint}/{parameter}").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:mainnet:sci:{index}/{entrypoint}").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
+            format!("did:ccd:mainnet:sci:{index}/{entrypoint}/").parse::<Method>()?,
+            target
+        );
+        assert_eq!(
```

Comment:
Is this assertion looks identical to the one above the one above.

Reply:
True. Not sure what the intention was.

Reply:
I removed it.